metabase/metabase
# add button to send diagnostic data to slack

Closes https://github.com/metabase/metabase/issues/49542

### Description

Add a button to send diagnostic data to slack. Right now, there’s an option to download the data as a json when cmd/ctrl + F1 key is pressed. View [this project](https://github.com/orgs/metabase/projects/102/views/1) for more details. This feature is only available on stats.metabase.com or localhost.

In order for the slack integration to work, `MB_SLACK_BOT_TOKEN` needs to be set up.

### How to verify

1. cmd/ctrl + F1
2. "send to slack" button appears
3. hit the button and see a report appear in #bug-reports-internal in Slack metabase

### Tests
- bug report channel input appears on Slack integration when MB_BUG_REPORTER_ENABLED is set
- bug report channel input field is not visible when MB_BUG_REPORTER_ENABLED is not set
- “send to slack” button appears when all of these are true:
    - bug report channel is set in slack settings
    - MB_BUG_REPORTER_ENABLED env variable is set up
- “send to slack” button does not appear when either one of the conditions is missing
- “send to slack” button shows loading 
- “send to slack” button hides loading when action is done
- notification message is shown when sending to slack is successful and unsuccessful
- diagnostic data can be downloaded
- diagnostic data is sent to files channel
- diagnostic data summary with original file attached is sent to bug reports channel
